### DISCUSSION TITLE
`latest` not generated for `type=semver,pattern={{raw}}`

### DISCUSSION THREAD

**Discussion by tisonkun** [docker/metadata-action#437] [Q&A]
My semver is `v0.1.0-alpha.1`. Does it mean that only formal release trigger semver?

It's undocumented at the README but reads like:

https://github.com/docker/metadata-action?tab=readme-ov-file#latest-tag

Every `type=semver,pattern=...` would generate `latest`.

**Comment by chaunsin** [comment]
> Every type=semver,pattern=... would generate latest.

I ran some tests here and it looks like this

**Reply by crazy-max** [reply]
It doesn't for rc, beta, alpha, ..., see https://github.com/docker/metadata-action?tab=readme-ov-file#typesemver

> *Pre-release (rc, beta, alpha) will only extend {{version}} (or {{raw}} if specified) as tag because they are updated frequently, and contain many breaking changes that are (by the author's design) not yet fit for public consumption.

But agree that we should explain it in the `Latest tag` section of the docs

**Comment by polarathene** [comment]
In the scope of only `type=semver` and the default `flavor.latest=auto` this question has been answered that the answer is yes unless the tag rule encounters a pre-release semver value.

If you have other tag types involved, the `priority` attribute and how the value to process is sourced can result in opt-out. [Covered in detail here](https://github.com/docker/metadata-action/issues/593).
